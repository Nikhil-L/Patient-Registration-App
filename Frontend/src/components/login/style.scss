
.login{
  background-image: url("../../cerner.jpg");
  width : 1280px;
  height: 609px;
  background-size: cover;
.base-container {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%,-50%);
  width: 350px;
  height: 400px;
  padding-top: 10px;
  box-sizing: border-box;
  background: rgba(0, 0, 0,.2);
  
    .content {

      .image {
        width: 12em;
        img {
          margin-left: 150px;
          margin-bottom: 40px;
          width: 100%;
          height: 100%;
          align-self: center;
          object-fit:cover;
          border-radius:50%;
        }
      }
  
      .form {
        margin-top: 0em;
        display: flex;
        flex-direction: column;
        align-items: center;
  
        .form-group {
          display: flex;
          flex-direction: column;
          align-items: flex-start;
          width: fit-content;
          .text-error{
            margin-left: 100px;
            color: red;
          }
          label {
            font-size: 20px;
          }
          input {
            margin-left:  100px;
            margin-top: 6px;
            min-width: 18em;
            height: 37px;
            padding: 0px 10px;
            font-size: 16px;
            font-family: "Open Sans", sans-serif;
            background-color: #f3f3f3;
            border: 0;
            border-radius: 4px;
            margin-bottom: 31px;
            
          }
        }
      }
    }
  
    .footer {
      margin-top: 1em;
      .btn{
        color : white;
        background-color: black;
        width : 100px;
        height: 35px;
        align-items: center;
        align-self: center;
        margin-left: 200px;
      }
    }
  }
}