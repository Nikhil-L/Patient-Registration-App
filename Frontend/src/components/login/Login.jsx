import React from "react";
import "./style.scss"
import loginImg from "../../logo.png";

/**
 * This component is used to handle all the login and its events
 */
export class Login extends React.Component {

  /**
    * Initializes the props
    * @param {*} props 
    */
  constructor(props) {
    super(props);

    this.state = {
      input: {},
      errors: {},
    };
     
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

  }

  /**
     * This method is used to handle the changes made on the webpage
     * @param {*} event 
     */
  handleChange(event) {
    let input = this.state.input;
    input[event.target.name] = event.target.value;
  
    this.setState({
      input
    });
  }

  /**
     * This method is used to handle the submissions
     * @param {*} event 
     */
  handleSubmit(event) {
    event.preventDefault();
    this.validate();
  }

  /**
     * This is to validate the authenticity of the user
     *
     */
  validate(){
    let url = 'http://localhost:8082/patient_registration_war/webapi/login';
    let postData = {
        "username":this.state.input["username"],
        "password":this.state.input["password"]
      };
    const axios = require('axios');
    axios.post(url, postData)
    .then((response)=>{ 
      console.log(response)
      sessionStorage.setItem("username", this.state.input["username"])
      sessionStorage.setItem("isAuthenticated", true)
      response.data?window.location="/home":this.setState(
      {errors:{"errormsg":"Invalid username or password"}})
    }).catch(error => {
      if(!error.response)
        window.location = "/dberror"
    });
}

 /**
  * This is used to render the component to the webpage
  * @returns a html code 
  */
  render() {
    
    return (
      <div className="login">
      <div className="base-container" ref={this.props.containerRef}>
        <form onSubmit={this.handleSubmit}>
          <div className="content">
            <div className="image">
              <img src={loginImg} />
            </div>
            <div className="form">
              <div className="form-group">
              <div className="text-error">{this.state.errors.errormsg}</div>
                <input 
                  data-testid="test-input"
                  type="text"
                  name="username"
                  value={this.state.input.username}
                  onChange={this.handleChange}
                  placeholder="username" />
                
              </div>
              <div className="form-group">
                <input
                  data-testid="test-input1"
                  type="password"
                  name="password"
                  value={this.state.input.password}
                  onChange={this.handleChange}
                  placeholder="password" />
              
              </div>
            </div>
          </div>
          <div className="footer">
            <button type="submit" value="submit" className="btn">
             <b>Login</b> 
            </button>
          </div>
        </form>
      </div>
      </div>
    );

  }
}
