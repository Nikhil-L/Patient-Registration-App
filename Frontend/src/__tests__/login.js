import React from "react";
import '@testing-library/jest-dom'
import {Login} from "../components/login/Login"
import renderer from "react-test-renderer"
import {shallow}  from "enzyme"


/**
 * This is to test the Login component
 */
describe("Login test", ()=>{

  /**
   * Snapshot testing
   */
  it("matches a snapshot", ()=>{
    const tree = renderer.create(<Login/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

/**
   * login event testing
   */
it('login check', ()=>{
  let wrapper = shallow(<Login/>);
  wrapper.find('input[type="text"]').simulate('change', {target: {name: 'username', value: 'krishankantsinghal'}});
  wrapper.update();
  expect(wrapper.find('input[type="text"]').props().value).toEqual('krishankantsinghal');
})

it('login end to end testing', ()=>{
  let wrapper = shallow(<Login/>);
  wrapper.find('form').props().onSubmit({preventDefault: jest.fn()});
  wrapper.find('input[type="text"]').simulate('change', {target:
    {name:'username', value:'Arun_kumar'}})
    wrapper.find('input[type="password"]').simulate('change', {target:
      {name:'password', value:'Ark@234'}})
    wrapper.find('form').props().onSubmit({preventDefault: jest.fn()});
})

})

