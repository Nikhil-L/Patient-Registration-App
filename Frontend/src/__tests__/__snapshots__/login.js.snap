// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test matches a snapshot 1`] = `
<div
  className="login"
>
  <div
    className="base-container"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="image"
        >
          <img
            src="cerner2.png"
          />
        </div>
        <div
          className="form"
        >
          <div
            className="form-group"
          >
            <div
              className="text-error"
            />
            <input
              data-testid="test-input"
              name="username"
              onChange={[Function]}
              placeholder="username"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              data-testid="test-input1"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
            />
          </div>
        </div>
      </div>
      <div
        className="footer"
      >
        <button
          className="btn"
          type="submit"
          value="submit"
        >
          <b>
            Login
          </b>
        </button>
      </div>
    </form>
  </div>
</div>
`;
